pkg/windows/calico.go

```diff
@@ -0,0 +1,549 @@
+	CalicoSystemNamespace  = "calico-system"
+	CalicoChart            = "rke2-calico"
+)
+
+// Setup creates the basic configuration required by the CNI.
+func (c *Calico) Setup(ctx context.Context, nodeConfig *daemonconfig.Node) (*CniConfig, error) {
+	// TODO:  Need to generate windows containerd to include CNI config
+	if err := setWindowsAgentSpecificSettings(nodeConfig); err != nil {
+		return nil, err
+	}
+
+	calicoKubeConfig := CalicoKubeConfig{
+		Server:               "https://127.0.0.1:6443",
+		CertificateAuthority: filepath.Join(GetWindowsAgentDir(), "server-ca.crt"),
+		Token:                "",
+		Path:                 filepath.Join(GetWindowsAgentDir(), CalicoKubeConfigName),
+	}
+
+	restConfig, err := clientcmd.BuildConfigFromFlags("", nodeConfig.AgentConfig.KubeConfigK3sController)
+	if err != nil {
+		return nil, err
+	}
+	client, err := coreClient(restConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	// TODO: kubectl get secret -n calico-system | grep calico-node
+	// Need to generate this on the server and send it down. Hmm really, maybe it should
+	// be in the Calico class and use a more generic kubeconfig.
+	secrets, err := client.CoreV1().Secrets(CalicoSystemNamespace).List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	for _, secret := range secrets.Items {
+		if !isCalicoNodeToken(&secret) {
+			continue
+		}
+		value, ok := secret.Data["token"]
+		if !ok {
+			continue
+		}
+		calicoKubeConfig.Token = string(value)
+	}
+	if calicoKubeConfig.Token == "" {
+		return nil, errors.New("could not retrieve calico node token from the cluster")
+	}
+
+	cfg := &CniConfig{NodeConfig: nodeConfig, NetworkName: "Calico", BindAddress: nodeConfig.AgentConfig.NodeIP}
+	if err := getDefaultConfig(cfg, nodeConfig); err != nil {
+		return nil, err
+	}
+
+	cfg.CalicoConfig.KubeConfig = calicoKubeConfig
+	if err := getCniConfigOverrides(cfg, restConfig); err != nil {
+		return nil, err
+	}
+
+	// TODO: Fix this to the package
+	//agent.SetHnsNetworkName(CalicoHnsNetworkName)
+
+	if err := createNodeNameFile(cfg); err != nil {
+		return nil, err
+	}
+	if err := createKubeConfig(cfg); err != nil {
+		return nil, err
+	}
+
+	if err := createCniConfig(cfg); err != nil {
+		return nil, err
+	}
+	return cfg, nil
+}
+
+// Start starts the CNI services on the Windows node.
+func (c *Calico) Start(config *CniConfig) error {
+	if err := generateCalicoNetworks(config.CalicoConfig.NetworkingBackend); err != nil {
+		return err
+	}
+
+	if err := startCalico(config.CalicoConfig); err != nil {
+		return err
+	}
+	// TODO: Test without it
+	time.Sleep(5 * time.Second)
+
+	if err := startFelix(config.CalicoConfig); err != nil {
+		return err
+	}
+	return nil
+}
+
+// createNodeNameFile creates the node name file required by the CNI.
+func createNodeNameFile(config *CniConfig) error {
+	return util2.WriteFile(filepath.Join(GetWindowsAgentDir(), CalicoNodeNameFileName), config.NodeConfig.AgentConfig.NodeName)
```

Comment:
I think you need to pass through the data-dir from `nodeConfig *daemonconfig.Node` and build the agent dir relative to that (there are existing functions for this), instead of getting hardcoded paths from`GetWindowsAgentDir()`